spark-jobs/src/main/scala/za/co/absa/enceladus/common/RecordIdGeneration.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.common
+
+import com.typesafe.config.{Config, ConfigException}
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions.callUDF
+import org.slf4j.{Logger, LoggerFactory}
+import za.co.absa.enceladus.utils.udf.{UDFLibrary, UDFNames}
+
+object RecordIdGeneration {
+
+  sealed trait UuidType
+
+  object UuidType {
+    case object TrueUuids extends UuidType
+    case object PseudoUuids extends UuidType
+    case object NoUuids  extends UuidType
+  }
+
+  private val log: Logger = LoggerFactory.getLogger(this.getClass)
+
+  def getRecordIdGenerationStrategyFromConfig(conf: Config): UuidType = {
+    val strategyValue = conf.getString("enceladus.recordid.generation.strategy")
+
+    strategyValue.toLowerCase match {
+      case "true" => UuidType.TrueUuids
+      case "pseudo" => UuidType.PseudoUuids
+      case "no" => UuidType.NoUuids
```

Comment:
**Question:** What about '' (empty string) meaning the same?

Reply:
My initial response is: no, why to introduce another value when the 3 are quite explanatory, while the empty string would not.

What is your motivation?

Reply:
Just considering user expectations. Nothing specified means no key generation required - to some users.
But if it feels weird, let's not do it. In case we would get errors due to that, it's easy to extend.